### ISSUE TITLE
genutil.statistics.mean does not exists

### ISSUE THREAD

**Issue by chaosphere2112** [UV-CDAT/genutil#5]
because MV2.average does pretty much the same thing

but this would allow to use nifty `axis='t'` or `axis='xy'` or `axis='(my_axis_name)'`

Migrated from: https://github.com/UV-CDAT/uvcdat/issues/1569

**Comment by durack1** [comment]
@jasonb5 this is related to the fact that cdms/cdutil etc all had their origin in numeric, before numpy and as numeric didn't include all the functionality that numpy now has, basic stats function were created and are now duplicated by the numpy variants, assuming numpy is wrapped correctly to expose all the functions. The MV2 library could mostly be deprecated in favour of numpy pure functions, or alternatively MV2 just calls the numpy functions directly (preserving the cdms transient variable attributes of course)

**Comment by jypeter** [comment]
I totally agree with accessing `numpy` (more likely `numpy.ma`) with MV2, in order to preserve attributes would be very useful!

It could be useful to add a new `compute_history` attribute to the variable specifying which function was applied, and append this if there is a pre-existing `compute_history` attribute (same way the `history` global attribute works in nc files)

Because in some cases, the original name/units of the variable may not mean anything after applying the numpy function

**Comment by durack1** [comment]
@jypeter another good idea, so similar to how `nco` operates with appending to the `history` global attribute with a summary of what was done.

I am sure @pochedls will also have some ideas around this

**Comment by jasonb5** [comment]
The `MV2` module already wraps numpy.ma functions. I see there is an `average` but not `mean` in `MV2`, I can add this.

As for history recording, we're planning on adding provenance support in the next-gen CDMS2.

**Comment by durack1** [comment]
@jasonb5 thanks for the update, when we talked about this a long time ago, I was suggesting that removing the duplication in MV2 by rather wrapping all the numpy.ma functions was a good idea, which means that if numpy.ma adds another function (or bugfix) we get this in CDMS for free with no maintenance overhead for you folks

**Comment by jasonb5** [comment]
Ah I see what you mean, this is inline with the next-gen CDMS2 where `MV2` will be absorbed into the `Variable` class.

**Comment by durack1** [comment]
Perfect, that sounds like we're on the same page here
